include/object/reply.h

```diff
@@ -11,12 +11,12 @@
 #include <object/structures.h>
 
 /* Unlink a reply from its tcb */
-static inline void reply_unlink(reply_t *reply)
+static inline void reply_unlink(reply_t *reply, tcb_t *tcb)
```

Comment:
It's fine to do this for the inline function as the compiler can optimise away the extra argument.

Why do we need to change the signatures of pop and remove though? If that can be avoided that would be great. Ideally we pass less arguments around than more. 

From an api perspective it's very confusing, as the signature of the function implies you can pass tcb and reply that are unlinked. 

That said, if it helps the proofs through and there's no easy way around it, go ahead.

Reply:
Thanks, Anna. I see what you mean by this being confusing in terms of API. It is true that this will speed up the proof significantly, and we do want to take advantage of that for now. I think this is something we can come back later for an improvement. I’ll create a JIRA issue for it.